src/HealthCheck/DbHealthCheck.php

```diff
@@ -0,0 +1,92 @@
+<?php
+declare(strict_types = 1);
+
+namespace ShoppinPal\YapepCommon\HealthCheck;
+
+use ShoppinPal\YapepCommon\Exception\HealthCheckException;
+use YapepBase\Database\DbConnection;
+use YapepBase\Database\DbFactory;
+use YapepBase\Database\MysqlConnection;
+use YapepBase\Exception\DatabaseException;
+use YapepBase\Exception\ParameterException;
+
+class DbHealthCheck implements IHealthCheck
+{
+    public function checkServiceHealth(array $configOptions = []): void
+    {
+        if (!isset($configOptions['connectionName'])) {
+            throw new ParameterException('Missing connectionName from config options for DB health check');
+        }
+
+        $connectionName = $configOptions['connectionName'];
+        $checkRo        = $configOptions['checkRo'] ?? true;
+        $checkRw        = $configOptions['checkRw'] ?? true;
+
+        if (!$checkRo && !$checkRw) {
+            throw new ParameterException(
+                'Neither RO or RW connection checks are enabled for DB health check ' . $connectionName
+            );
+        }
+
+        // Make sure there are no existing connections in the DB factory
+        DbFactory::clearConnections();
+
+        // RO must be checked first, as after an RW connection, the RW connection will always be returned
+        if ($checkRo) {
+            $this->performCheck($connectionName, DbFactory::TYPE_READ_ONLY);
+        }
+
+        if ($checkRw) {
+            $this->performCheck($connectionName, DbFactory::TYPE_READ_WRITE);
+        }
+    }
+
+    /**
+     * @throws HealthCheckException
+     */
+    protected function performCheck(string $connectionName, string $connectionType): void
+    {
+        try {
+            $errorPrefix = 'Error while creating connection';
```

Comment:
This is a bit pervert :)

Reply:
I'd create a HealthCheckAbstract class with abstract methods:
validateConfig()
performCheck()

and the abstract class can run these in order. As I saw there are little difference between the check classes

Reply:
I really don't want to over-engineer this. The idea is we should be able to easily add custom health checks for other dependencies, and what the structure of those checks is is completely up to the person implementing them.

The abstract class would just add a more rigid structure for this without really any benefit.